functions/common/debezium-autoconfigure/README.adoc

```diff
@@ -0,0 +1,252 @@
+# Debezium Auto-Configuration
+
+This module provides a generic https://debezium.io/documentation/reference/2.2/development/engine.html[DebeziumEngine.Builder] auto-configuration that can be reused and composed in other applications.
+
+NOTE: `DebeziumEngine` doesn't not required Kafka or Kafka Connect as it runs embedded inside your application. This approach though comes with some delivery guarantee limitations as explained https://debezium.io/documentation/reference/2.2/development/engine.html#_handling_failures[here].
+
+The `Debezium Engine` is a https://en.wikipedia.org/wiki/Change_data_capture[Change Data Capture] (CDC) utility, that allows *capturing* database change events and process them with custom `java.util.Consumer` or `io.debezium.engine.ChangeConsumer` event handler implementations. The `java.util.Consumer` allow element-wise change event processing while the `ChangeConsumer` permits batch event processing.
+
+The `DebeziumEngine.Builder` auto-configuration is activated only if a Debezium Connector is found on the classpath and the `debezium.properties.connector.class` property is set.
+
+To process the incoming database change events you need to include the Debezium `auto-configuration` dependency to your project:
+
+====
+[source, xml, subs="normal", role="primary"]
+.Maven
+----
+<dependency>
+    <groupId>org.springframework.cloud.fn</groupId>
+    <artifactId>debezium-autoconfigure</artifactId>
+    <version>{project-version}</version>
+</dependency>
+----
+[source, groovy, subs="normal", role="secondary"]
+.Gradle
+----
+compile "org.springframework.cloud.fn:debezium-autoconfigure:{project-version}"
+----
+====
+
+As well as the https://debezium.io/documentation/reference/2.2/connectors/index.html[debezium connector] dependency for the Database the data is received from.
+For example for PostgreSQL the postgres debezium connector dependency is:
+
+====
+[source, xml, subs="normal", role="primary"]
+.Maven
+----
+<dependency>
+    <groupId>io.debezium</groupId>
+    <artifactId>debezium-connector-postgres</artifactId>
+    <version>${debezium-version}</version>
+</dependency>
+
+----
+[source, groovy, subs="normal", role="secondary"]
+.Gradle
+----
+compile "io.debezium:debezium-connector-postgres:{debezium-version}"
+----
+====
+
+
+Then implement your own `java.util.Consumer<ChangeEvent>` (or `ChangeConsumer<ChangeEvent>`) handler bean and wire it into a DebeziumEngine instance. For example:
```

Comment:
I don't think it is really a strict requirement to have such a consumer as a bean.
There is just enough to have a `DebeziumEngine.Builder` autowired, set some consumer into it and build an engine.
There is even no requirement to have that engine as a bean as well.
There is probably just enough to have something like `MyDebeziumService` marked with a `@Service` implementing a `SmartLifecycle`, and having that auto-configured `DebeziumEngine.Builder` injected.

Sure! It is possible what you show so far, but don't make end-users stuck just with only this style 😄

Reply:
Good point. I've added some related notes under the sample. 
Maybe we can think of different or another sample to ilustrate it.